pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -312,6 +315,34 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 			condemned = append(condemned, pods[i])
 		}
 		// If the ordinal could not be parsed (ord < 0), ignore the Pod.
+
+		for _, requested := range set.Spec.VolumeClaimTemplates {
+			pvcName := getPersistentVolumeClaimName(set, &requested, getOrdinal(pods[i]))
+			actual, err := ssc.client.CoreV1().PersistentVolumeClaims(set.Namespace).Get(pvcName, metav1.GetOptions{})
+			if err != nil {
+				glog.Warningf("Couldn't find a volume matching template %s on pod %s: %v",
+					requested.Name,
+					pods[i].Name,
+					err,
+				)
+				status.VolumeClaimTemplateMismatches++
+				continue
+			}
+
+			// TODO: Compare more properties? Storage class, labels, ...
+			requestedStorage := requested.Spec.Resources.Requests[v1.ResourceStorage]
```

Comment:
Rather than just indicating a mismatch, consider adding a condition that indicates weather the volume is under provisioned or over provisioned. Also consider adding a condition for storage class mismatch.